### PR TITLE
8 packages from smimram/ocaml-pandoc at 0.2.0

### PR DESCRIPTION
**Pull request by smimram**
This pull-request concerns:
- `pandoc.0.2.0`: Library to write pandoc filters
- `pandoc-abbreviations.0.2.0`: Pandoc filter to add non-breaking spaces after abbreviations
- `pandoc-comment.0.2.0`: Pandoc filter to remove comments
- `pandoc-crossref.0.2.0`: Pandoc filter to have LaTeX cross-references
- `pandoc-include.0.2.0`: Pandoc filter to include other files
- `pandoc-inspect.0.2.0`: Pandoc filter to inspect pandoc's JSON
- `pandoc-pdf2png.0.2.0`: Pandoc filter to change extension of pdf images to png
- `pandoc-replace.0.2.0`: Pandoc filter to replace words

---
* Homepage: https://github.com/smimram/ocaml-pandoc
* Source repo: git+https://github.com/smimram/ocaml-pandoc.git
* Bug tracker: https://github.com/smimram/ocaml-pandoc/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/pandoc/pandoc.0.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+description:
+  "This library helps writing filters for pandoc, which is a tool to convert between textual formats notably, markdown and LaTeX"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "yojson" {>= "1.4.0"}
```
Comment (avsm) [inline packages/pandoc/pandoc.0.2.0/opam:13]:
```suggestion
  "dune" {>= "2.0"}
  "ocaml" {>= "4.08.0"}
  "yojson" {>= "2.0.0"}
```

fix CI failures for syntax and lower bounds

File: packages/pandoc/pandoc.0.2.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+license: "LGPL-2.1"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.08.0"}
+  "yojson" {>= "2.0.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (avsm) [inline packages/pandoc/pandoc.0.2.0/opam:17]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-replace/pandoc-replace.0.2.0/opam [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (avsm) [inline packages/pandoc-replace/pandoc-replace.0.2.0/opam:16]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-abbreviations/pandoc-abbreviations.0.2.0/opam [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to add non-breaking spaces after abbreviations"
+description:
+  "This pandoc filter adds non-breaking spaces after the abbreviations listed in the `abbreviations` file"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (avsm) [inline packages/pandoc-abbreviations/pandoc-abbreviations.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/pandoc-abbreviations/pandoc-abbreviations.0.2.0/opam [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+license: "LGPL-2.1-or-later"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (avsm) [inline packages/pandoc-abbreviations/pandoc-abbreviations.0.2.0/opam:17]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-crossref/pandoc-crossref.0.2.0/opam [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to have LaTeX cross-references"
+description:
+  "This pandoc filter replaces #chap: and #sec: references to cref invokations"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (avsm) [inline packages/pandoc-crossref/pandoc-crossref.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/pandoc-include/pandoc-include.0.2.0/opam [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to include other files"
+description:
+  "This pandoc filter allows the inclusion of other markdown files into markdown"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (avsm) [inline packages/pandoc-include/pandoc-include.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/pandoc-include/pandoc-include.0.2.0/opam [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (avsm) [inline packages/pandoc-include/pandoc-include.0.2.0/opam:16]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-crossref/pandoc-crossref.0.2.0/opam [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1-or-later"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (avsm) [inline packages/pandoc-crossref/pandoc-crossref.0.2.0/opam:16]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-pdf2png/pandoc-pdf2png.0.2.0/opam [hunk 10]
```diff
@@ -0,0 +1,37 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to change extension of pdf images to png"
+description:
+  "This pandoc filter changes the extension of images from pdf to png."
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (avsm) [inline packages/pandoc-pdf2png/pandoc-pdf2png.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/pandoc/pandoc.0.2.0/opam [hunk 11]
```diff
@@ -0,0 +1,38 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Library to write pandoc filters"
+description:
+  "This library helps writing filters for pandoc, which is a tool to convert between textual formats notably, markdown and LaTeX"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (avsm) [inline packages/pandoc/pandoc.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/pandoc-replace/pandoc-replace.0.2.0/opam [hunk 12]
```diff
@@ -0,0 +1,37 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to replace words"
+description:
+  "This pandoc filter replaces words by others (useful to ensure that the case is respected for instance)."
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (avsm) [inline packages/pandoc-replace/pandoc-replace.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

**Comment by mseri** [comment]
As a rule of thumb, to simplify maintenance of the metadata, we recommend to use strict `=version` bounds for packages that are released as a bundle. So the `pandoc-*` packages would all use `"pandoc" {=version}` as constraint

**Comment by smimram** [comment]
Thanks for the feedback this will be changed in future versions of the package.

File: packages/pandoc-abbreviations/pandoc-abbreviations.0.2.0/opam [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+  "This pandoc filter adds non-breaking spaces after the abbreviations listed in the `abbreviations` file"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1-or-later"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04.0"}
+  "pandoc" {>= "0.1.0"}
```
Comment (mseri) [inline packages/pandoc-abbreviations/pandoc-abbreviations.0.2.0/opam:14]:
```suggestion
  "pandoc" {= version}
```

File: packages/pandoc-include/pandoc-include.0.2.0/opam [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+description:
+  "This pandoc filter allows the inclusion of other markdown files into markdown"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1-or-later"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
```
Comment (mseri) [inline packages/pandoc-include/pandoc-include.0.2.0/opam:13]:
```suggestion
  "pandoc" {>= "0.2.0"}
```

File: packages/pandoc-replace/pandoc-replace.0.2.0/opam [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+description:
+  "This pandoc filter replaces words by others (useful to ensure that the case is respected for instance)."
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1-or-later"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
```
Comment (mseri) [inline packages/pandoc-replace/pandoc-replace.0.2.0/opam:13]:
```suggestion
  "pandoc" {>= "0.2.0"}
```

**Comment by raphael-proust** [comment]
> Thanks for the feedback this will be changed in future versions of the package.

Should we mark the depencies as `>= "0.2.0" & < "0.3.0"` to avoid having to fix these packages at the next release? Or just use `= version` right now to get the same benefits?

Also, https://github.com/ocaml/opam-repository/pull/24796 is a fix for the versions 0.1.0 so they don't install concurrently with 0.2.0.

**Comment by avsm** [comment]
Merged #24796 and just running one more CI before merging.

File: packages/pandoc-comment/pandoc-comment.0.2.0/opam [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/pandoc-comment/pandoc-comment.0.2.0/opam:16]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-inspect/pandoc-inspect.0.2.0/opam [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/pandoc-inspect/pandoc-inspect.0.2.0/opam:16]:
```suggestion
  ["dune" "subst"] {dev}```

File: packages/pandoc-pdf2png/pandoc-pdf2png.0.2.0/opam [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1-or-later"
+homepage: "https://github.com/smimram/ocaml-pandoc"
+bug-reports: "https://github.com/smimram/ocaml-pandoc/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "pandoc" {>= "0.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/pandoc-pdf2png/pandoc-pdf2png.0.2.0/opam:16]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/pandoc-inspect/pandoc-inspect.0.2.0/opam [hunk 19]
```diff
@@ -0,0 +1,37 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to inspect pandoc's JSON"
+description:
+  "This pandoc filter prints pandoc's internal JSON on the standard error"
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (mseri) [inline packages/pandoc-inspect/pandoc-inspect.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

File: packages/pandoc-comment/pandoc-comment.0.2.0/opam [hunk 20]
```diff
@@ -0,0 +1,37 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pandoc filter to remove comments"
+description:
+  "This pandoc filter removes LaTeX-style comments (line starting with %)."
+maintainer: ["Samuel Mimram <smimram@gmail.com>"]
+authors: ["Samuel Mimram <smimram@gmail.com>"]
+license: "LGPL-2.1"
```
Comment (mseri) [inline packages/pandoc-comment/pandoc-comment.0.2.0/opam:8]:
```suggestion
license: "LGPL-2.1-or-later"
```

**Comment by mseri** [comment]
Thanks! Please upstream the changes to the opam files

**Pull request merged**